connector/server/handlers/medco_node.go

```diff
@@ -106,5 +104,119 @@ func MedCoNodeExploreQueryHandler(params medco_node.ExploreQueryParams, principa
 			EncryptedPatientList: query.Result.EncPatientList,
 			Timers:               timers,
 			Status:               models.ExploreQueryResultElementStatusAvailable,
+			PatientSetID:         int64(query.Result.PatientSetID),
 		}})
 }
+
+// MedCoNodeGetCohortsHandler handles GET /medco/node/explore/cohorts  API endpoint
+func MedCoNodeGetCohortsHandler(params medco_node.GetCohortsParams, principal *models.User) middleware.Responder {
+	userID := principal.ID
+	cohorts, err := querytoolsserver.GetSavedCohorts(userID)
+	if err != nil {
+		medco_node.NewGetCohortsDefault(500).WithPayload(&medco_node.GetCohortsDefaultBody{
+			Message: "Get cohort execution error: " + err.Error(),
+		})
+	}
+	payload := &medco_node.GetCohortsOK{}
+	for _, cohort := range cohorts {
+		payload.Payload = append(payload.Payload,
+			&medco_node.GetCohortsOKBodyItems0{
+				CohortName:   cohort.CohortName,
+				CohortID:     int64(cohort.CohortID),
+				QueryID:      int64(cohort.QueryID),
+				CreationDate: cohort.CreationDate.Format(time.RFC3339),
+				UpdateDate:   cohort.UpdateDate.Format(time.RFC3339),
+			},
+		)
+	}
+
+	return medco_node.NewGetCohortsOK().WithPayload(payload.Payload)
+
+}
+
+// MedCoNodePostCohortsHandler handles POST /medco/node/explore/cohorts  API endpoint
+func MedCoNodePostCohortsHandler(params medco_node.PostCohortsParams, principal *models.User) middleware.Responder {
+
+	cohort := params.Body
+
+	hasID, err := querytoolsserver.CheckQueryID(principal.ID, int(cohort.PatientSetID))
+	if err != nil {
+		return medco_node.NewPostCohortsDefault(500).WithPayload(&medco_node.PostCohortsDefaultBody{
+			Message: fmt.Sprintf("During execution of CheckQueryID"),
+		})
+	}
+	logrus.Trace("has ID", hasID)
+
+	if !hasID {
+		return medco_node.NewPostCohortsDefault(400).WithPayload(&medco_node.PostCohortsDefaultBody{
```

Comment:
The 400 response should be done with the proper generated wrapper, which should be something like `NewPostCohortsBadRequest` (generated only if it specified in the swagger file).